pkg/chaosdaemon/server.go

```diff
@@ -159,45 +154,70 @@ type RegisterGatherer interface {
 	prometheus.Gatherer
 }
 
-// StartServer starts chaos-daemon.
-func StartServer(conf *Config, reg RegisterGatherer) error {
-	g := &errgroup.Group{}
+// Server is the server for chaos daemon
+type Server struct {
+	errors       chan error
+	daemonServer *DaemonServer
 
-	httpBindAddr := conf.HttpAddr()
-	httpServer := newHTTPServerBuilder().Addr(httpBindAddr).Metrics(reg).Profiling(conf.Profiling).Build()
+	conf *Config
+	reg  RegisterGatherer
+}
+
+// BuildServer builds a chaos daemon server
+func BuildServer(conf *Config, reg RegisterGatherer) (*Server, error) {
+	daemonServer, err := newDaemonServer(conf.Runtime, reg)
+	if err != nil {
+		return nil, errors.Wrap(err, "create daemon server")
+	}
 
-	grpcBindAddr := conf.GrpcAddr()
+	return &Server{
+		errors:       make(chan error),
+		daemonServer: daemonServer,
+		conf:         conf,
+		reg:          reg,
+	}, nil
+}
+
+// Start starts chaos-daemon.
+func (s *Server) Start() error {
```

Comment:
How about building `Start()` like `http.Server`'s `ListenAndServe()`? Do not use `Errors() <-chan error`.

Reply:
How about errgroup?

Reply:
Fixed